RevenueCatUI/PaywallView.swift

```diff
@@ -259,7 +209,42 @@ private extension PaywallView {
         }
     }
 
-    static func loadCachedCurrentOfferingIfPossible() -> Offering? {
+    func loadOffering() async throws -> Offering {
+        switch self.contentToDisplay {
+        case let .offering(offering):
+            return offering
+
+        case .defaultOffering:
+            return try await Purchases.shared.offerings().current.orThrow(PaywallError.noCurrentOffering)
+
+        case let .offeringIdentifier(identifier):
+            return try await Purchases.shared.offerings()
+                .offering(identifier: identifier)
+                .orThrow(PaywallError.offeringNotFound(identifier: identifier))
+
+        case let .offeringPlacementIdentifier(placementIdentifier):
+            return try await Purchases.shared.offerings()
+                .currentOffering(forPlacement: placementIdentifier)
+                .orThrow(PaywallError.offeringNotFound(identifier: placementIdentifier))
+        }
+    }
+
+}
+
+// MARK: -
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private extension PaywallViewConfiguration.Content {
+
+    func extractInitialOffering() -> Offering? {
+        switch self {
+        case let .offering(offering): return offering
+        case .defaultOffering: return Self.loadCachedCurrentOfferingIfPossible()
+        case .offeringIdentifier, .offeringPlacementIdentifier: return nil
```

Comment:
this is probably wrong.
If we have a cached one - we should return it.
Otherwise you'd always see a progress for a split second

Reply:
I see this is how they have implemented it 😮‍💨 , I guess it's fine then

Reply:
`offeringPlacementIdentifier` has been added by me, and this was my same doubt, but I preferred to follow the same approach they had with `offeringIdentifier `.